File: app/src/main/java/com/alexpaxom/workrhythm/ui/main/PageFragment.java
package com.alexpaxom.workrhythm.ui.main;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;


import androidx.lifecycle.LiveData;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.alexpaxom.workrhythm.R;
import com.alexpaxom.workrhythm.model.Task;
import com.alexpaxom.workrhythm.viewmodel.TaskStorageViewModel;

import java.util.List;


/**
 * A page fragment containing a simple view.
 */
public class PageFragment extends Fragment {

    private TaskStorageViewModel viewModel;
    private ListTasksAdapter listTasksAdapter;
    private int layout_view;
    private Integer shown_status;

    public PageFragment() {
        this(R.layout.tab_task_list_in_queue, Task.IN_QUEUE);
    }

    public PageFragment(int layout_view, int shown_status) {
        this.layout_view = layout_view;
        this.shown_status = shown_status;
    }


    @Override
    public void onCreate(Bundle savedInstanceState) {
        viewModel = new ViewModelProvider(requireActivity()).get(TaskStorageViewModel.class);
        super.onCreate(savedInstanceState);
    }


    // Inflate the view for the fragment based on layout XML
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(layout_view, container, false);

        RecyclerView rwListTasks;

        rwListTasks = (RecyclerView) view.findViewById(R.id.list_tasks);
        listTasksAdapter = new ListTasksAdapter(this, shown_status);
        LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity());
        rwListTasks.setLayoutManager(layoutManager);
        rwListTasks.setAdapter(listTasksAdapter);

        LiveData<List<Task>> liveDataTasksList = viewModel.getTasks(shown_status);

        liveDataTasksList.observe(getViewLifecycleOwner(), new Observer<List<Task>>() {
            @Override
            public void onChanged(List<Task> taskList) {
                listTasksAdapter.reloadListData(taskList);
            }
        });

        return view;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        if(listTasksAdapter != null)
            listTasksAdapter.onDestroyAdapter();
    }
}

File: app/src/main/java/com/alexpaxom/workrhythm/app/fragments/PageFragment.kt
package com.alexpaxom.workrhythm.app.fragments

import kotlin.jvm.JvmOverloads
import com.alexpaxom.workrhythm.R
import com.alexpaxom.workrhythm.app.viewmodel.TaskStorageViewModel
import com.alexpaxom.workrhythm.app.adapters.ListTasksAdapter
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.LinearLayoutManager
import com.alexpaxom.workrhythm.data.model.Task

/**
 * A page fragment containing a simple view.
 */
class PageFragment @JvmOverloads constructor(
    private val layout_view: Int = R.layout.tab_task_list_in_queue,
    private val shown_status: Int = Task.IN_QUEUE
) : Fragment() {
    private var viewModel: TaskStorageViewModel? = null
    private var listTasksAdapter: ListTasksAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        viewModel = ViewModelProvider(requireActivity()).get(
            TaskStorageViewModel::class.java
        )
        super.onCreate(savedInstanceState)
    }

    // Inflate the view for the fragment based on layout XML
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(layout_view, container, false)
        val rwListTasks: RecyclerView
        rwListTasks = view.findViewById<View>(R.id.list_tasks) as RecyclerView
        listTasksAdapter = ListTasksAdapter(this, shown_status)
        val layoutManager = LinearLayoutManager(activity)
        rwListTasks.layoutManager = layoutManager
        rwListTasks.adapter = listTasksAdapter
        val liveDataTasksList = viewModel!!.getTasks(
            shown_status
        )
        liveDataTasksList!!.observe(
            viewLifecycleOwner,
            { taskList -> listTasksAdapter!!.reloadListData(taskList as List<Task>) })
        return view
    }

    override fun onDestroyView() {
        super.onDestroyView()
        if (listTasksAdapter != null) listTasksAdapter!!.onDestroyAdapter()
    }
}